### PR TITLE
Google services

### PR DESCRIPTION
**Pull request by swebdevelopment**
Please review this pull request. 
- Implemented new Google SignIn for Gmail, Drive and Photos

### CODE REVIEW

File: FSPicker/Platforms/iOS/Models/FSGoogleServicesManager.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
+//
+//  FSGoogleServicesManager.h
+//  Pods
+//
+//  Created by Alexanedr on 4/18/17.
+//  Copyright (c) 2017 __MyCompanyName__. All rights reserved.
+//
+
```
Comment (kyuss) [inline FSPicker/Platforms/iOS/Models/FSGoogleServicesManager.h:8]: Can you please remove this?

File: FSPicker/Platforms/iOS/Models/FSGoogleServicesManager.m [hunk 2]
```diff
@@ -0,0 +1,80 @@
+//
+//  FSGoogleServicesManager.m
+//  Pods
+//
+//  Created by Alexanedr on 4/18/17.
+//  Copyright (c) 2017 __MyCompanyName__. All rights reserved.
+//
```
Comment (kyuss) [inline FSPicker/Platforms/iOS/Models/FSGoogleServicesManager.m]: Can you please remove this?

File: GoogleServicesIntegration.rtf [hunk 3]
```diff
@@ -0,0 +1,49 @@
+{\rtf1\ansi\ansicpg1252\cocoartf1404\cocoasubrtf470
```
Comment (kyuss) [inline GoogleServicesIntegration.rtf]: Can we have this file in markdown (.md) and referenced from README.md?

File: FSPicker/Platforms/iOS/API/FSUploader.m [hunk 4]
```diff
@@ -87,6 +103,254 @@ - (void)uploadCloudItems:(NSArray<FSContentItem *> *)items {
...
+                             }];
+        
+    }
+}
+
+
+- (NSString*)convertedGoogleMIMEType:(NSString*)mimeType{
+
+    if ([mimeType isEqualToString:@"application/vnd.google-apps.document"]) {
+        //return @"application/vnd.oasis.opendocument.text";
```
Comment (kyuss) [inline FSPicker/Platforms/iOS/API/FSUploader.m:318]: Remove commented code

File: FSPicker/Platforms/iOS/API/FSUploader.m [hunk 5]
```diff
@@ -87,6 +103,254 @@ - (void)uploadCloudItems:(NSArray<FSContentItem *> *)items {
...
+        return @"application/vnd.openxmlformats-officedocument.wordprocessingml.document";
+    }
+    
+    if ([mimeType isEqualToString:@"application/vnd.google-apps.spreadsheet"]) {
+        //return @"application/x-vnd.oasis.opendocument.spreadsheet";
```
Comment (kyuss) [inline FSPicker/Platforms/iOS/API/FSUploader.m:323]: Remove commented code

File: FSPicker/Platforms/iOS/Models/FSContentItem.m [hunk 6]
```diff
@@ -26,18 +26,170 @@ - (instancetype)initWithDictionary:(NSDictionary *)dictionary {
...
     for (NSDictionary *item in content) {
         FSContentItem *contentItem = [[FSContentItem alloc] initWithDictionary:item];
         [items addObject:contentItem];
     }
+    
+    return items;
+}
+
++ (NSArray<FSContentItem *> *)itemsFromGTLRDriveFileList:(GTLRDrive_FileList *)fileList {
+    //NSArray<NSDictionary *> *content = [[NSArray alloc] initWithArray:json[@"contents"]];
```
Comment (kyuss) [inline FSPicker/Platforms/iOS/Models/FSContentItem.m]: Remove commented code

File: FSPicker/Platforms/iOS/Models/FSGoogleServicesManager.m [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+- (void)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+    
+    NSError* configureError;
+    [[GGLContext sharedInstance] configureWithError: &configureError];
+    NSAssert(!configureError, @"Error configuring Google services: %@", configureError);
+    
+//    [GIDSignIn sharedInstance].delegate = self;
+//    [GIDSignIn sharedInstance].scopes = @[@"https://www.googleapis.com/auth/drive"];
+
```
Comment (kyuss) [inline FSPicker/Platforms/iOS/Models/FSGoogleServicesManager.m]: Remove commented code

**Comment by philolo1** [comment]
@swebdevelopment @kyuss any progress on this, is this fixing https://github.com/filestack/FSPicker/issues/25 maybe?

**Comment by rnine** [comment]
@swebdevelopment I was reviewing your code today and I found a number of issues when trying to use it in 2 test apps. Let me summarize:

1. On a brand new Objective-C app I was (apparently) able to authenticate against Google Drive using my Google credentials, but then I am presented with an empty list of files (there should be folders and files there) and the logout button is grayed out. Same applies when authenticating against Gmail.

2. On a brand new Swift 3 app I can not install the `FPPicker` pod anymore due to the fact that one of the new dependencies: `Google/SignIn`, includes static binaries. Unfortunately CocoaPods does not support these when `use_frameworks!` is enabled, which is required when using CocoaPods with Swift. See https://github.com/CocoaPods/CocoaPods/issues/6036.

3. CocoaPods reports that the `Google` pod is deprecated when I run `pod install` on the Objective-C project I mentioned before:

```
[!] Google has been deprecated
```

To be honest, their documentation is a bit confusing on this respect and I am not entirely sure what's the replacement for it.

As you can see, there's a few concerns that should be addressed before this PR can be merged into the repository and we would love to hear your input on this!

**Comment by Uniprog** [comment]
Hello @rnine 
Thanks for your feedback. I have few questions/comments:
1. Did you use steps from GoogleServicesIntegration.md to integrate FSPicker to test apps? Could you please double check steps 7 and 10 from GoogleServicesIntegration.md. By your description, looks like app has issue with API permissions or issue with configuration in GoogleService-Info.plist.
2. I'm working on solution for Swift 3 and CocoaPods
3. Google pod v3.1.0 was deprecated. We still could use it, but I will look for some fix/workaround.

**Comment by rnine** [comment]
@Uniprog

1. Oops, you were right, I forgot to enable the “Drive API” and “Gmail API” in the Google console. It is working fine now! Only issue I can notice is that the logout button does not seem to log the user out from Google services. Is this something you could improve?

2 and 3: OK, thanks! Looking forward to your solution/workaround 👍

**Pull request merged**